src/group_theory/sylow.lean

```diff
@@ -17,21 +17,146 @@ The Sylow theorems are the following results for every finite group `G` and ever
+    one_mem' := ⟨Q, ⟨⟨Q, hQ⟩, rfl⟩, Q.one_mem⟩,
+    inv_mem' := λ g ⟨_, ⟨R, rfl⟩, hg⟩, ⟨R, ⟨R, rfl⟩, R.1.inv_mem hg⟩,
+    mul_mem' := λ g h ⟨_, ⟨R, rfl⟩, hg⟩ ⟨_, ⟨S, rfl⟩, hh⟩, (hc2.total_of_refl R.2 S.2).elim
+      (λ T, ⟨S, ⟨S, rfl⟩, S.1.mul_mem (T hg) hh⟩) (λ T, ⟨R, ⟨R, rfl⟩, R.1.mul_mem hg (T hh)⟩) },
+  λ ⟨g, _, ⟨S, rfl⟩, hg⟩, by
+  { refine exists_imp_exists (λ k hk, _) (hc1 S.2 ⟨g, hg⟩),
+    rwa [subtype.ext_iff, coe_pow] at hk ⊢ },
+  λ M hM g hg, ⟨M, ⟨⟨M, hM⟩, rfl⟩, hg⟩⟩) P hP) (λ Q ⟨hQ1, hQ2, hQ3⟩, ⟨⟨Q, hQ1, hQ3⟩, hQ2⟩)
+
+instance sylow.nonempty : nonempty (sylow p G) :=
+nonempty_of_exists is_p_group.of_bot.exists_le_sylow
+
+noncomputable instance sylow.inhabited : inhabited (sylow p G) :=
+classical.inhabited_of_nonempty sylow.nonempty
+
+open_locale pointwise
+
+/-- `subgroup.pointwise_mul_action` preserves Sylow subgroups. -/
+instance sylow.pointwise_mul_action {α : Type*} [group α] [mul_distrib_mul_action α G] :
+  mul_action α (sylow p G) :=
+{ smul := λ g P, ⟨g • P, P.2.map _, λ Q hQ hS, inv_smul_eq_iff.mp (P.3 (hQ.map _)
+    (λ s hs, (congr_arg (∈ g⁻¹ • Q) (inv_smul_smul g s)).mp
+      (smul_mem_pointwise_smul (g • s) g⁻¹ Q (hS (smul_mem_pointwise_smul s g P hs)))))⟩,
+  one_smul := λ P, sylow.ext (one_smul α P),
+  mul_smul := λ g h P, sylow.ext (mul_smul g h P) }
+
+instance sylow.mul_action : mul_action G (sylow p G) :=
+mul_action.comp_hom _ mul_aut.conj
+
+lemma sylow.coe_smul {g : G} {P : sylow p G} :
+  ↑(g • P) = P.1.map (mul_aut.conj g).to_monoid_hom := rfl
+
+lemma sylow.smul_eq_iff_mem_normalizer {g : G} {P : sylow p G} :
+  g • P = P ↔ g ∈ P.1.normalizer :=
+begin
+  rw [eq_comm, sylow.ext_iff, set_like.ext_iff, ←inv_mem_iff, mem_normalizer_iff, inv_inv],
+  exact forall_congr (λ h, iff_congr iff.rfl ⟨λ ⟨a, b, c⟩, (congr_arg _ c).mp
+    ((congr_arg (∈ P.1) (mul_aut.inv_apply_self G (mul_aut.conj g) a)).mpr b),
+    λ hh, ⟨(mul_aut.conj g)⁻¹ h, hh, mul_aut.apply_inv_self G (mul_aut.conj g) h⟩⟩),
+end
+
+lemma subgroup.sylow_mem_fixed_points_iff (H : subgroup G) {P : sylow p G} :
+  P ∈ fixed_points H (sylow p G) ↔ H ≤ P.1.normalizer :=
+by simp_rw [set_like.le_def, ←sylow.smul_eq_iff_mem_normalizer]; exact subtype.forall
+
+lemma is_p_group.inf_normalizer_sylow {P : subgroup G} (hP : is_p_group p P) (Q : sylow p G) :
+  P ⊓ Q.1.normalizer = P ⊓ Q :=
+le_antisymm (le_inf inf_le_left (sup_eq_right.mp (Q.3 (hP.to_inf_left.to_sup_of_normal_right'
+  Q.2 inf_le_right) le_sup_right))) (inf_le_inf_left P le_normalizer)
+
+lemma is_p_group.sylow_mem_fixed_points_iff
+  {P : subgroup G} (hP : is_p_group p P) {Q : sylow p G} :
+  Q ∈ fixed_points P (sylow p G) ↔ P ≤ Q :=
+by rw [P.sylow_mem_fixed_points_iff, ←inf_eq_left, hP.inf_normalizer_sylow, inf_eq_left]
+
+/-- A generalization of **Sylow's second theorem**.
+  If the number of Sylow `p`-subgroups is finite, then all Sylow `p`-subgroups are conjugate. -/
+lemma sylow.conjugate [hp : fact p.prime] [fintype (sylow p G)] (P Q : sylow p G) :
+  ∃ g : G, g • P = Q :=
```

Comment:
I think there might be a typeclass for this?

Reply:
My guess would be `transitive`, but it doesn't seem to exist under that name.

Reply:
Pretty close, it's `mul_action.is_pretransitive` I think?

Reply:
I've added an instance

Reply:
IMO you should just remove `sylow.conjugate` and provide the instance directly